parsl/executors/high_throughput/process_worker_pool.py

```diff
@@ -590,28 +590,28 @@ def update_resource_spec_env_vars(mpi_launcher: str, resource_spec: Dict, node_i
         os.environ[key] = prefix_table[key]
 
 
-def execute_task(bufs, mpi_launcher: Optional[str] = None):
-    """Deserialize the buffer and execute the task.
+def _init_mpi_env(mpi_launcher: str, resource_spec: Dict):
+    node_list = resource_spec.get("MPI_NODELIST")
+    if node_list is None:
+        return
+    nodes_for_task = node_list.split(',')
+    logger.info(f"Launching task on provisioned nodes: {nodes_for_task}")
+    update_resource_spec_env_vars(mpi_launcher=mpi_launcher, resource_spec=resource_spec, node_info=nodes_for_task)
+
 
+def execute_task(bufs: bytes):
+    """Deserialize the buffer and execute the task.
     Returns the result or throws exception.
     """
     user_ns = locals()
     user_ns.update({'__builtins__': __builtins__})
 
-    f, args, kwargs, resource_spec = unpack_res_spec_apply_message(bufs, user_ns, copy=False)
+    f, args, kwargs, resource_spec = unpack_res_spec_apply_message(bufs, copy=False)
```

Comment:
As you've just identified that `user_ns` isn't used here, might as well move the creation of `user_ns` closer to where it's needed, and then can remove one of the `.update()` calls.

Reply:
Good call.

Reply:
Agreed; I can do it in a follow up PR.